src/cardsort/analysis.py

```diff
@@ -13,19 +14,47 @@
 ]
 
 
-def _get_distance_matrix_for_user(df_user: pd.DataFrame) -> np.ndarray:
-    n = len(df_user)
+logger = logging.getLogger(__name__)
 
-    X = np.zeros((n, n))
-    for i in range(n):
-        for j in range(i, n):
-            cat1 = df_user.loc[df_user["card_id"] == (i + 1), "category_label"].values[
-                0
-            ]
-            cat2 = df_user.loc[df_user["card_id"] == (j + 1), "category_label"].values[
-                0
-            ]
-            X[i, j] = X[j, i] = 0 if cat1 == cat2 else 1
+
+def _check_data(df: pd.DataFrame) -> bool:
+    # check if user_ids go from 1 to n=number of users
+    user_ids = df["user_id"].unique()
+    for i, user_id in enumerate(user_ids):
+        if user_id != i + 1:
+            logger.error("User_ids are not in the right order or contain gaps.")
+            return False
+    # check if each card_id is always associated with exactly one card_label
+    card_id_counts = df.groupby("card_id")["card_label"].nunique()
+    for card_id, count in card_id_counts.items():
+        if count != 1:
+            logger.error(
+                f"Card_id {card_id} is associated with {count} different card_labels."
+            )
+            return False
+    # check if card_ids go from 1 to n=number of cards
+    card_ids = df["card_id"].unique()
```

Comment:
With the new implementation of the distance matrix, there is no more need for the card_ids to go from 1 to n=number of cards, this check can be removed

Reply:
Done: c9f27d7